modules/gitops-storing-aws-secret-manager-resources-in-gitops-repository.adoc

```diff
@@ -0,0 +1,181 @@
+// Module is included in the following assemblies:
+//
+// * securing_openshift_gitops/managing-secrets-securely-using-sscsid-with-gitops.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="gitops-storing-aws-secret-manager-resources-in-gitops-repository_{context}"]
+= Storing AWS Secrets Manager resources in {gitops-shortname} repository
+
+This guide provides instructions with examples to help you use {gitops-shortname} workflows with the Secrets Store Container Storage Interface (SSCSI) Driver Operator to mount secrets from AWS Secrets Manager to a CSI volume in OpenShift Container Platform.
+
+[IMPORTANT]
+====
+It is not supported to use the SSCSI Driver Operator with AWS Secrets Manager in a hosted control plane cluster.
+==== 
+
+.Prerequisites
+
+* You have access to the cluster with `cluster-admin` privileges.
+* You have access to the {OCP} web console.
+* You have extracted and prepared the `ccoctl` binary.
+* You have installed the `jq` CLI tool.
+* Your cluster is installed on AWS and uses AWS Security Token Service (STS).
+* You have configured AWS Secrets Manager to store the required secrets.
+* link:https://docs.openshift.com/container-platform/latest/nodes/pods/nodes-pods-secrets-store.html#persistent-storage-csi-secrets-store-driver-install_nodes-pods-secrets-store[SSCSI Driver Operator is installed on your cluster].
+* {gitops-title} Operator is installed on your cluster.
+* You have a {gitops-shortname} repository ready to use the secrets.
+* You are logged in to the Argo CD instance by using the Argo CD admin account.
+
+.Procedure
+
+. Install the AWS Secrets Manager provider and add resources:
+
+.. In your {gitops-shortname} repository, create a directory and add `aws-provider.yaml` file in it with the following configuration to deploy resources for the AWS Secrets Manager provider:
++
+[IMPORTANT]
+====
+The AWS Secrets Manager provider for the SSCSI driver is an upstream provider.
+
+This configuration is modified from the configuration provided in the upstream link:https://github.com/aws/secrets-store-csi-driver-provider-aws#installing-the-aws-provider[AWS documentation] so that it works properly with {OCP}. Changes to this configuration might impact functionality.
+====
++
+.Example `aws-provider.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: csi-secrets-store-provider-aws
+  namespace: openshift-cluster-csi-drivers
+---
```

Comment:
There is an unwanted horizontal line, can we avoid that?
![Screenshot from 2023-12-20 12-55-40](https://github.com/openshift/openshift-docs/assets/60093147/3d3a2995-0ee0-4aa6-8b34-afb68afdd969)

Reply:
@varshab1210  Thanks for asking a very good question. This usage of three hyphens (---) is to separate YAML definitions in a single YAML file. Hence retaining this as is.

Refer to the RH OCP Doc guidelines: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#yaml-formatting